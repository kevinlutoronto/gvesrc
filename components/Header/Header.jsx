// @flow
import cx from 'classnames';
import React from 'react';
import style from './style.scss';
import { getDataAttrs } from '../../dataUtils';
import type { Data, StyleAttributes } from '../../types';

type Props = {|
  text?: string,
  children?: React$Node,
  underline?: boolean,
  h1?: boolean,
  h2?: boolean,
  h3?: boolean,
  h4?: boolean,
  h5?: boolean,
  h6?: boolean,
  invert?: boolean,
  style?: StyleAttributes,
  noMargin?: boolean,
  data?: Data,
|};

class Header extends React.Component<Props> {
  render() {
    const headerStyle = cx(style.header, {
      [style.header1]: this.props.h1,
      [style.header2]: this.props.h2,
      [style.header3]: this.props.h3,
      [style.header4]: this.props.h4,
      [style.header5]: this.props.h5,
      [style.header6]: this.props.h6,
      [style.invert]: this.props.invert,
      [style.noMargin]: this.props.noMargin,
    });

    return (
      <div {...getDataAttrs(this.props.data)} className={[headerStyle]}>
        <div className={style.content}>{this.props.text || this.props.children}</div>
        {this.props.underline && <div className={style.rule} />}
      </div>
    );
  }
}

Header.displayName = 'Plasma@Header';

export default Header;
