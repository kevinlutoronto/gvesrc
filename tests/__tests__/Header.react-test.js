/* eslint-disable */

import React from 'react';
import Header from '../../components/Header/Header';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Header Component', () => {
  test('render', () => {
    const component = renderer.create(<Header>YOLO</Header>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
