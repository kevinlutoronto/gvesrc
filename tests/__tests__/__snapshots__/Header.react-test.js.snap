// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component render 1`] = `
<div
  className={
    Array [
      "header",
    ]
  }
>
  <div
    className="content"
  >
    YOLO
  </div>
</div>
`;
